src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -163,24 +165,51 @@ public static double BitIncrement(double x)
             return BitConverter.Int64BitsToDouble(bits);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe double CopySign(double x, double y)
         {
-            // This method is required to work for all inputs,
-            // including NaN, so we operate on the raw bits.
+            const long signMask
+                = unchecked((long)0b_1000_0000__0000_0000__0000_0000__0000_0000__0000_0000__0000_0000__0000_0000__0000_0000);
```

Comment:
That's some constant 😆 

Surely `1L << 63` is more readable?

Reply:
...that is a very fair point, not sure why i chose this way 😂